App/Containers/AddCaptionScreen.js

```diff
@@ -1,110 +1,79 @@
 import React from 'react'
 import { connect } from 'react-redux'
-import { View, Text, Image, ScrollView } from 'react-native'
+import { View } from 'react-native'
 import { TextileHeaderButtons, Item } from '../Components/HeaderButtons'
 import Input from '../SB/components/Input'
-import { Button } from 'react-native-elements'
 import { NavigationActions } from 'react-navigation'
 import styles from '../SB/views/ThreadCreate/statics/styles'
 import UIActions from '../Redux/UIRedux'
-import CameraRollActions from '../Redux/CameraRollRedux'
-
 
 class AddCaptionScreen extends React.Component {
-  constructor (props) {
-    super(props)
-    this.state = {
-      value: ''
-    }
-  }
-
   static navigationOptions = ({ navigation }) => {
     const params = navigation.state.params || {}
     return {
       headerTitle: 'Photo Caption',
       headerLeft: (
         <TextileHeaderButtons left>
-          <Item title='Back' iconName='arrow-left' onPress={() => { navigation.dispatch(NavigationActions.back()) }} />
+          <Item title='Back' iconName='arrow-left' onPress={() => {
+            params.cancelShare()
+            navigation.dispatch(NavigationActions.navigate({ routeName: params.backTo }))
```

Comment:
This is cool. With the new version of `react-navigation` we updated to recently, you can just `navigate` to a route name and not worry about if that route is already in the stack. If it's not, it will be pushed onto the stack. If it is in the stack, it will navigate 'back' to it.